admin_guide/cluster_management.rst

```diff
@@ -64,37 +63,43 @@ Citus can easily tolerate worker node failures because of its logical sharding-b
 
     WARNING:  could not connect to node localhost:9700
 
-On seeing such warnings, the first step would be to remove the failed worker from the pg_worker_list.conf file in the data directory.
+On seeing such warnings, the first step would be to log into the failed node and
+inspect the cause of the failure.
 
-::
+Citus will automatically re-route the work to the healthy workers. Also, if Citus is not able to connect to a worker, it will assign that task to another node having a copy of that shard. If the failure occurs mid-query, Citus does not re-run the whole query but assigns only the failed query fragments leading to faster responses in face of failures.
 
-    vi $PGDATA/pg_worker_list.conf
+Once the node is brought back up, Citus will automatically continue connecting to it and
+using the data. 
 
-.. note::
-    The instruction above assumes that the data directory is in the PGDATA environment variable. If not, you will need to set it. For example:
-    
-    ::
-        
-        export PGDATA=/usr/lib/postgresql/9.5/data
+If the node suffers a permanent failure then you may wish to retain the same
+level of replication so that your application can tolerate more failures. To
+make this simpler, Citus enterprise provides a replicate_table_shards UDF which
+can be called after. This function copies the shards of a table across the
+healthy nodes so they all reach the configured replication factor.
 
-Then, you can reload the configuration so that the master picks up the desired configuration changes.
+First, you should mark all shard placements on that node as invalid (if they are
```

Comment:
To remove a permanently failed node from the list of workers, you should first mark .......